[Steven Chua] iP

Graphcalibur opened the pull request:
# MNSKY

> "I can help!" - MNSKY, Fargo Season 3

MNSKY can help you free your mind from all the things you need to remember! They're

- text-based (for now!)
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is

1. download it from [here](https://github.com/Graphcalibur/ip/releases).
2. double-click the JAR file.
3. add your tasks.
4. let MNSKY help you 😉

And it is **FREE**! ~~But feel free to send me money~~

Features:

- [x]     Managing tasks
- [x]     Managing deadlines
- [ ]     Reminders (coming soon)
- [ ]     Start the inevitable robot uprising
- [x]     Can help!

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Mnsky mnsky = new Mnsky();
        mnsky.run();
    }
}
```

Commit:
Add Gradle support

Graphcalibur committed:
Add Greeting, Echo and Exit functions to MNSKY

Graphcalibur committed:
Rename main file and class from Duke to MNSKY

Graphcalibur committed:
Add list functionality

Graphcalibur committed:
Changed class and variable names of Mnsky to follow Java conventions

Graphcalibur committed:
Add Task class as well as mark and unmark functionality

Graphcalibur committed:
Add support for tracking ToDos, Events, and Deadlines

Graphcalibur committed:
Add automated text UI testing

Graphcalibur committed:
Update gitignore

Graphcalibur committed:
Add comments and Exception handling

Graphcalibur committed:
Add Delete functionality

Graphcalibur committed:
Change some println() calls to printf()

Graphcalibur committed:
Add saving and loading the list to memory.

Graphcalibur committed:
Add support for date and time

Graphcalibur committed:
Add extra support for date and time

Graphcalibur committed:
Fix merge conflicts

Graphcalibur committed:
Add Storage object to handle file I/O

Graphcalibur committed:
Improve exception handling

Graphcalibur committed:
Add Ui object to handle user interaction

Graphcalibur committed:
Add Ui object to handle user interaction

Graphcalibur committed:
Remove data folder and updated gitignore

Graphcalibur committed:
Add Parser object

Graphcalibur committed:
Added TaskList object

Graphcalibur committed:
Add Packages

Graphcalibur committed:
Add JUnit tests

Graphcalibur committed:
Add META_INF for jar file

Graphcalibur committed:
Add javadoc comments

Graphcalibur committed:
Update code to follow coding standards

Graphcalibur committed:
Fix switch statements

Graphcalibur committed:
Fix bugs

Graphcalibur committed:
Add find functionality

Graphcalibur committed:
Fix merge conflicts

Graphcalibur committed:
Fix merge conflicts

Graphcalibur committed:
Fix bugs

Graphcalibur committed:
Merge branch 'add-gradle-support' of https://github.com/Graphcalibur/ip into branch-A-Gradle

Graphcalibur committed:
Fix test bugs in TaskListTest

Graphcalibur committed:
Edit gradle build file to run Mnsky class

Graphcalibur committed:
Add checkstyle and fixed coding standard violations

Graphcalibur committed:
Add JavaFX to gradle build dependencies

Graphcalibur committed:
Add GUI

Graphcalibur committed:
Fix MNSKY's long responses getting cut off

Graphcalibur committed:
Add and update Javadoc comments

Graphcalibur committed:
Refactor code and close window on bye command

seanleong339 commented on src/main/java/mnsky/Mnsky.java:
```diff
@@ -0,0 +1,108 @@
+
+            try {
+                switch (parsedInput.get(0)) {
+                case "bye":
```
Good job on not making common mistake of indenting before case! 👍

seanleong339 commented on src/main/java/mnsky/Parser.java:
```diff
@@ -0,0 +1,190 @@
+import mnsky.exceptions.MnskyException;
+import mnsky.exceptions.MnskyMissingParameterException;
+
+public class Parser {
```
Perhaps you could add a header comment to this class? 🛩️ 
I noticed this in some other classes as well

seanleong339 commented on src/main/java/mnsky/TaskList.java:
```diff
@@ -0,0 +1,255 @@
+                    break;
+
+                default:
+                    throw new MnskyException("????");
```
Perhaps the Exception message could have more details for easier debugging?

Graphcalibur replied:
This exception is intended to never be thrown but I think I could make the message more helpful

seanleong339 commented on src/main/java/mnsky/TaskList.java:
```diff
@@ -0,0 +1,255 @@
+     */
+    public Task addTask(String name) {
+        Task task = new Task(name);
+        tasks.add(task);
```
Perhaps renaming the ArrayList variable as "tasklist" or "list" instead of "tasks" would be clearer? 😄

Graphcalibur replied:
Not sure, I called it tasks instead of taskList so that it wouldn't be confused with the actual taskList object itself. The coding standard also mentioned that lists of objects should be named as plural nouns.

seanleong339 reviewed (commented):
Overall, good comments on the method! I can understand what the classes and methods do 💯 I've put some suggestions just for you to consider, but this would be more of preference.

Graphcalibur committed:
Change look of GUI

jbkim1999 commented on src/main/java/mnsky/core/Mnsky.java:
```diff
@@ -0,0 +1,98 @@
+
+        try {
+            ArrayList<String> parsedInput = Parser.parseInput(input);
+            switch (parsedInput.get(0)) {
```
I like how you followed the Java coding standards, by having no indentation for case clauses!

jbkim1999 commented on src/main/java/mnsky/core/TaskList.java:
```diff
@@ -0,0 +1,260 @@
+                break;
+
+            default:
+                throw new MnskyException("????");
```
Could you perhaps add a more meaningful error message for this particular exception? (It may not be executed after all, but it could still provide a better insight to the viewers I think!)

jbkim1999 commented on src/main/java/mnsky/task/Event.java:
```diff
@@ -0,0 +1,67 @@
+
+        String dateString = "";
+        String timeString = "";
+        boolean dateExists = atDate != null;
```
I like the use of your boolean names!

jbkim1999 commented on src/main/java/mnsky/task/Task.java:
```diff
@@ -0,0 +1,62 @@
+package mnsky.task;
+
```
Could you perhaps add a header Javadoc comment for this class? I also noticed some subclasses of this class have no header comments as well.

jbkim1999 reviewed (commented):
Overall, I think you did really well in terms of following the coding standards. Only a very few minor glitches to fix.

Graphcalibur committed:
Add some assertions

Graphcalibur committed:
Refactor code to improve quality

Code has some magic numbers and convoluted statements.

This makes the code more difficult and confusing to read.

Let's
* Replace the magic numbers with constant variables with
descriptive names
* Extract out complicated statements into multiple lines

This way, the meaning of the magic numbers will be understood.
Having complicated statements spread out onto multiple lines
makes them easier to read and understand.

Graphcalibur committed:
Add undo and redo functionality

Changes made to the list are permanent and have to be manually
undone or redone if the user wants to undo or redo them.

This makes changing the list tedious if the user makes a mistake.

Let's add undo and redo functionality by using two stacks to keep
track of the history of the tasks ArrayList in the history
taskList.

This is the simplest way to allow users to undo and redo any
changes made to the taskList.

Graphcalibur committed:
Refactor code

Graphcalibur committed:
Add greeting message from MNSKY on startup

Graphcalibur committed:
Update readme

Graphcalibur committed:
Merge pull request #1 from Graphcalibur/branch-C-Undo

Add Undo and Redo functionality

Graphcalibur committed:
Merge branch 'master' into branch-A-Assertions

Graphcalibur committed:
Merge pull request #2 from Graphcalibur/branch-A-Assertions

Add some assertions

Graphcalibur committed:
Merge branch 'master' into branch-A-CodeQuality

Graphcalibur committed:
Merge pull request #3 from Graphcalibur/branch-A-CodeQuality

Refactor code to improve quality

Graphcalibur committed:
Improve GUI

Graphcalibur committed:
Fix bug with retrieving Storage data

Graphcalibur committed:
Fix bug with undo and redo

Graphcalibur committed:
Set theme jekyll-theme-hacker

Graphcalibur committed:
Add user guide

Graphcalibur committed:
Fix minor error in user guide

Graphcalibur committed:
Merge branch 'master' of https://github.com/Graphcalibur/ip

Graphcalibur committed:
Improve GUI

Graphcalibur committed:
Update Ui image

Graphcalibur committed:
Fix minor issue

Graphcalibur committed:
Edit build.gradle

Graphcalibur committed:
Fix issue with text wrapping

Graphcalibur committed:
Improve bye functionality

Graphcalibur committed:
Update user guide

Graphcalibur committed:
Update user guide

Graphcalibur committed:
Add table of contents to user guide

Graphcalibur committed:
Update table of contents in user guide

Graphcalibur committed:
Add CI

Graphcalibur committed:
Fix errors in testing
